### PR TITLE
Move kvs access to DaemonCursorStorage

### PR DESCRIPTION
**Pull request by johannkm**
I need something like this to support the AssetDaemon in cloud where the kvs is on cloud storage instead of run storage, and we've talked about moving in this direction anyway

https://github.com/dagster-io/internal/pull/5427

### CODE REVIEW

**Comment by johannkm** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13537** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13537" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13537?utm_source=stack-comment).

**Comment by prha** [comment]
This is more about creating a strict interface / tests for cloud than it is about cleaning up OSS, right?  Because we still need the `supports_kvs` flags and the run storage implementations for everything?

Are you thinking about splitting out the daemon cursor storage tests from the run storage tests?

**Comment by johannkm** [comment]
@prha yeah it's just to unify the interface for oss and cloud. The name is the only part that's getting stricter, vs having `instance.kvs` which I think would be even more appealing for arbitrary use

**Comment by prha** [comment]
While we're at it, can we extract kv tests out into its own abstract test class?  In OSS, we would just run it on `run_storage` instance.  But we could ensure the same behavior / same test suite between OSS and cloud and get rid of the standalone `test_kv.py` in cloud.

We might even be able to get rid of the `supports_kvs` flag, which I think is just to disable the run storage tests.

**Comment by johannkm** [comment]
separated the tests and removed supports_kvs

File: python_modules/dagster/dagster/_core/storage/daemon_cursor.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from abc import abstractmethod
+from typing import Mapping, Set
+
+
+class DaemonCursorStorage:
+    @abstractmethod
+    def kvs_get(self, keys: Set[str]) -> Mapping[str, str]:
```
Comment (gibsondan) [inline python_modules/dagster/dagster/_core/storage/daemon_cursor.py:7]: get_cursor_values?

File: python_modules/dagster/dagster/_core/storage/daemon_cursor.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+        """Retrieve the value for a given key in the current deployment."""
+
+    @abstractmethod
```
Comment (gibsondan) [inline python_modules/dagster/dagster/_core/storage/daemon_cursor.py:10]: set_cursor_values

**Review by gibsondan — APPROVED** [review]
excellent

**Review by prha — APPROVED** [review]
nicely done

**Pull request merged**